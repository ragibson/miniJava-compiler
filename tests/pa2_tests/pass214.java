// PA2 pass qualified reference
class A {
    void p(){
        A x = this.p();
    }
}
