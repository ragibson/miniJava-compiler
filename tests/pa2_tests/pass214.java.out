********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "A" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "p" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [1]
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     ClassType
  .   .   .       "A" Identifier
  .   .   .     "x" varname
  .   .   .   CallExpr
  .   .   .     QualRef
  .   .   .       "p" Identifier
  .   .   .       ThisRef
  .   .   .     ExprList + [0]
=============================================
